pontoon/base/models.py

```diff
@@ -89,17 +90,40 @@ def translations_count(query=None):
         )
 
 
-# User class extensions
+@property
+def user_translated_locales(self):
+    locales = []
+
+    for group in self.groups.all():
+        if hasattr(group, 'translated_locales'):
+            locale = group.translated_locales.first()
+            if locale:
+                locales.append(locale)
+        if hasattr(group, 'managed_locales'):
+            locale = group.managed_locales.first()
+            if locale:
+                locales.append(locale)
+
+    return [l.code for l in set(locales)]
+
+
+@property
+def user_display_permissions(self):
```

Comment:
This is display logic and I think would do better as a template helper function or filter.